File: MockInterview.Backend/MockInterview.BLL/Models/DTOs/InterviewDto.cs
using MockInterview.BLL.Models.Common;
using MockInterview.Core.Models.Entities;

namespace MockInterview.BLL.Models.DTOs;

/// <summary>
/// Represents one time interview
/// </summary>
public class InterviewDto : IEntityDto
{
    public long Id { get; set; }
    
    /// <summary>
    /// Interview record created date
    /// </summary>
    public DateTime CreatedDate { get; set; }
    
    /// <summary>
    /// Interview record updated date
    /// </summary>
    public DateTime UpdatedDate { get; set; }
    
    /// <summary>
    /// Decided time
    /// </summary>
    public DateTime InterviewTime { get; set; }

    /// <summary>
    /// Current status
    /// </summary>
    public string Status { get; set; } = null!;
    
    /// <summary>
    /// Related interviewer projection
    /// </summary>
    public InterviewerDto Interviewer { get; set; } = null!;

    /// <summary>
    /// Related interviewee projection
    /// </summary>
    public IntervieweeDto Interviewee { get; set; } = null!;
}

File: MockInterview.Frontend/src/types/interviewDto.ts
import {InterviewerDto} from "./interviewerDto";
import {IntervieweeDto} from "./intervieweeDto";
import {IMappable} from "./common";

export interface IInterviewDto {
    id: number;
    createdDate: Date;
    updatedDate: Date;
    interviewTime: Date;
    status: string;
    interviewer: InterviewerDto;
    interviewee: IntervieweeDto;
}

export class InterviewDto implements IMappable<IInterviewDto> {
    id!: number;
    createdDate!: Date;
    updatedDate!: Date;
    interviewTime!: Date;
    status!: string;
    interviewer!: InterviewerDto | null;
    interviewee!: IntervieweeDto | null;

    mapFrom(data: IInterviewDto) {
        this.id = data.id;
        this.createdDate = data.createdDate;
        this.updatedDate = data.updatedDate;
        this.interviewTime = data.interviewTime;
        this.status = data.status;
        this.interviewer = data.interviewer;
        this.interviewee = data.interviewee;
    }
}